tests: drivers: gpio: gpio_basic_api: Device runtime pm support for gpio basic api

smalae opened the pull request:
1. Implement device runtime PM for Si91x GPIO ports.
2. Add device runtime PM compatibility for gpio_basic_test

smalae committed:
drivers: gpio: siwx91x: Implement device runtime PM for GPIO ports

The existing device runtime PM model requires applications to call
pm_device_runtime_get()/put() on the GPIO controller device directly,
which is not ideal when GPIO ports are exposed as child nodes.

This update enables runtime PM at the GPIO port (child node) level,
allowing applications to manage power through the port devices instead
of the top-level controller.

Signed-off-by: Sai Santhosh Malae <Santhosh.Malae@silabs.com>

jerome-pouiller commented on tests/drivers/gpio/gpio_basic_api/testcase.yaml:
```diff
@@ -142,3 +142,9 @@ tests:
     extra_args: "DTC_OVERLAY_FILE=boards/frdm_rw612_hsgpio.overlay"
     harness_config:
       fixture: hsgpio_loopback
+  drivers.gpio.siwx91x_device_runtime_pm:
```
I think this test is not specific to siwx91x. You could name it `drivers.gpio.pm`.

smalae committed:
tests: drivers: gpio: Add device runtime PM compatibility

Update the gpio_basic_api test to support device runtime
power management. This ensures the test runs correctly when
runtime PM is enabled on GPIO devices.

Signed-off-by: Sai Santhosh Malae <Santhosh.Malae@silabs.com>

asmellby commented on tests/drivers/gpio/gpio_basic_api/src/main.c:
```diff
@@ -77,21 +78,33 @@ static void board_setup(void)
-
+	(void)pm_device_runtime_get(DEVICE_DT_GET(DEV));
+	if (IS_ENABLED(CONFIG_UART_CONSOLE)) {
+		(void)pm_device_runtime_get(DEVICE_DT_GET(DT_CHOSEN(zephyr_console)));
```
This won't work if `CONFIG_UART_CONSOLE=n` and `zephyr,console` is _not_ defined in DTS, as the macro expansion will fail due to missing intermediate macros.

This `if` must be wrapped in `#if DT_HAS_CHOSEN(zephyr_console)`.
